jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriterTest.java

```diff
@@ -0,0 +1,121 @@
+package com.exasol.adapter.dialects.bigquery;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import java.sql.*;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.*;
+import org.junit.jupiter.params.*;
+import org.junit.jupiter.params.provider.*;
+import org.mockito.*;
+import org.mockito.junit.jupiter.*;
+import org.mockito.quality.*;
+
+import com.exasol.*;
+import com.exasol.adapter.*;
+import com.exasol.adapter.dialects.*;
+import com.exasol.adapter.jdbc.*;
+import com.exasol.adapter.sql.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class BigQueryQueryRewriterTest extends AbstractQueryRewriterTest {
+    private QueryRewriter queryRewriter;
+    @Mock
+    private ResultSet mockResultSet;
+    @Mock
+    private ResultSetMetaData mockResultSetMetaData;
+    @Mock
+    private Statement mockStatement;
+    @Mock
+    private ExaMetadata exaMetadata;
+
+    @BeforeEach
+    void beforeEach() throws SQLException {
+        final Connection connectionMock = this.mockConnection();
+        this.statement = mock(SqlStatement.class);
+        final SqlDialect dialect = new BigQuerySqlDialect(connectionMock, AdapterProperties.emptyProperties());
+        final BaseRemoteMetadataReader metadataReader = new BaseRemoteMetadataReader(connectionMock,
+                AdapterProperties.emptyProperties());
+        this.queryRewriter = new BigQueryQueryRewriter(dialect, metadataReader, connectionMock);
+        when(connectionMock.createStatement()).thenReturn(this.mockStatement);
+        when(this.mockResultSet.getMetaData()).thenReturn(this.mockResultSetMetaData);
+        when(this.mockStatement.executeQuery(any())).thenReturn(this.mockResultSet);
+        when(this.mockResultSet.next()).thenReturn(true, false);
+        when(this.mockResultSetMetaData.getColumnCount()).thenReturn(1);
+    }
+
+    @Test
+    void testRewriteWithJdbcConnectionEmptyTable() throws AdapterException, SQLException {
+        when(this.mockResultSet.next()).thenReturn(false);
```

Comment:
If you have different `when`s, don't override it here, make sure each test has only what it needs. And that means removing anything that is not common to all test methods from `beforeTest()`.

Reply:
fixed